diff --git a/libc/Android.bp b/libc/Android.bp
index 6031b16..875adfe 100644
--- a/libc/Android.bp
+++ b/libc/Android.bp
@@ -53,6 +53,7 @@ libc_common_flags = [
     // Clang's exit-time destructor registration hides __dso_handle, but
     // __dso_handle needs to have default visibility on ARM32. See b/73485611.
     "-Wexit-time-destructors",
+    "-fno-stack-protector",
 ]
 
 // Define some common cflags
