diff --git a/init/service.cpp b/init/service.cpp
index a5334f4..560c122 100644
--- a/init/service.cpp
+++ b/init/service.cpp
@@ -371,6 +371,7 @@ bool Service::Start() {
             scon = ret_scon;
             free(ret_scon);
         }
+#if 0
         if (rc == 0 && scon == mycon) {
             ERROR("Service %s does not have a SELinux domain defined.\n", name_.c_str());
             if (selinux_status_getenforce() > 0) {
@@ -379,6 +380,7 @@ bool Service::Start() {
                 return false;
             }
         }
+#endif
         free(mycon);
         free(fcon);
         if (rc < 0) {
