diff --git a/services/sensorservice/SensorService.cpp b/services/sensorservice/SensorService.cpp
index 1c3e94354..e4c58d066 100644
--- a/services/sensorservice/SensorService.cpp
+++ b/services/sensorservice/SensorService.cpp
@@ -803,7 +803,10 @@ bool SensorService::threadLoop() {
     } while (!Thread::exitPending());
 
     ALOGW("Exiting SensorService::threadLoop => aborting...");
+#ifdef SENSOR_SERVICE_ABORT
     abort();
+#endif
+
     return false;
 }

